Controllers/ShiftsController.cs

```diff
@@ -45,53 +39,65 @@ public IEnumerable<Shift> GetAll()
         /// <response code="404">If the item is null</response>
         // GET api/<ShiftsController>/5
         [HttpGet("{id}")]
-        public ActionResult<Shift> Get(int id)
+        public async Task<IActionResult> Get(int id)
         {
-            var value = _randomizer.GenerateRandomData<Shift>(id);
-
-            if (value == null)
-            {
-                return NotFound();
-            }
-
-            return value;
+            var shift = await _context.Shifts.SingleOrDefaultAsync(shift => shift.id == id);
+            return shift == null ? NotFound() : Ok(shift);
         }
 
         /// <summary>
         /// Posts specific shift to Shift table
         /// </summary>
-        /// <param name="value">new created shift</param>
+        /// <param name="shift">new created shift</param>
         /// <returns>one shift by id</returns>
         /// <response code="201">Returns the newly created item</response>
         /// <response code="404">If the item is null</response>
         // POST api/<ShiftsController>
         [HttpPost]
-        public ActionResult<Shift> Post([FromBody] Shift value)
+        public async Task<IActionResult> Post([FromBody] CreateShiftDTO shift)
```

Comment:
The createdBy part is left out. Although, there is no validation and it is a bit pointless, we check for createdby in other parts of the system, so I think createBy field should be required and checked if it exists in BusinessManagers table.